oaknational/Oak-Web-Application
# Feat/pupil 1099/send page view events

### Job Story

**When** I access a page with share functionality,
**I want to send events to posthog**
**So that I can see the results of the experiment**.

### Implementation details

Put as much information here as possible about how to do this task

[https://www.notion.so/oaknationalacademy/Share-tracking-plan-13f26cc4e1b180379bf6c664fd120a77?pvs=4](https://www.notion.so/Share-tracking-plan-13f26cc4e1b180379bf6c664fd120a77?pvs=21)

- ShareExperimentPageAccessed (when user accesses page and feature flag is present)
- ShareExperimentConverted ( when user accesses page via url with extra params)
- store a cookie with the share-id
- Don’t send either event if the exisitng cookie matches

### Acceptance criteria

- [x]  ShareExperimentPageAccessed
- [x]  ShareExperimentConverted
- [x]  lesson overview browse & canonical
- [x] download confirmation
- [x] lesson listing
- [x]  duplicate events are not sent

## How to test

1. Go to https://deploy-preview-3012--oak-web-application.netlify.thenational.academy and accept cookies and open the console
2. Go to https://deploy-preview-3012--oak-web-application.netlify.thenational.academy/teachers/lessons/transverse-waves
3. See that TeacherShareInitiated is sent
4. Copy the url into a different browser window (eg. a different google instance or safari)
5. See that both TeacherShareIniated and TeacherShareConverted are sent (accept cookies)
6. See that the url in the second browser is modified to a different id
7. Repeat the experiment with same urls and see that the events are not sent the second time
8. repeat for the various pages
9. try with feature flags disabled in either browser and note that conversions are still sent will initiated events are not
